tests/test_tabulation.py

```diff
@@ -120,3 +120,332 @@ def runTest(self):
+        x = np.array([1, 2, 3])
+        y = np.array([4, 5, 6])
+        tab = Tabulation(x, y)
+        result = tab._update_y(None)
+        self.assertIs(result, tab)  # Should return the original object
+
+        # Test update with new_y having a different size than x
+        x = np.array([1, 2, 3])
+        y = np.array([4, 5, 6])
+        tab = Tabulation(x, y)
+        new_y = np.array([7, 8])  # Mismatched size
+        with self.assertRaises(ValueError) as context:
+            tab._update_y(new_y)
+        self.assertEqual(str(context.exception), "x and y arrays do not have the same size")
+
+        # Test xmerge with non-overlapping domains
+        x1 = np.array([1, 2, 3])
+        x2 = np.array([4, 5, 6])
+        with self.assertRaises(ValueError) as context:
+            result = Tabulation._xmerge(x1, x2)
+        self.assertEqual(str(context.exception), "domains do not overlap")
+
+        # resample where x=None
+        x = np.array([1, 2, 3])
+        y = np.array([4, 5, 6])
+
+        tab = Tabulation(x,y)
+
+        resampled = tab.resample(None)
+
+        self.assertTrue(np.all(resampled.x == x))
+        self.assertTrue(np.all(resampled.y == y))
+
+        # bandwidth_rms with dx=None
+        boxcar = Tabulation((0.,10.),(1.,1.))
+        value = 5
+
+        self.assertTrue(np.abs(boxcar.bandwidth_rms() - value) == 0.)
+
+        # Test multiplication of Two Tabulations
+        x1 = np.array([1, 2, 3])
+        y1 = np.array([4, 5, 6])
+        tab1 = Tabulation(x1, y1)
+
+        x2 = np.array([1, 2, 3])
+        y2 = np.array([1, 2, 3])
+        tab2 = Tabulation(x2, y2)
+
+        result = tab1 * tab2
+
+        expected_x = [1, 2, 3] # Merged x values
+
+        self.assertTrue(np.array_equal(result.x, expected_x))
+        self.assertTrue(np.array_equal(result(x), tab1(x) * tab2(x)))
+        self.assertTrue(np.array_equal(result.y, tab1.y * tab2.y))
+
+        # Test multiplication of Two Tabulations with a scalar
+        x = np.array([1, 2, 3])
+        y = np.array([4, 5, 6])
+        tab = Tabulation(x, y)
+
+        scalar = 2
+
+        result = tab * scalar
+
+        assert np.array_equal(result(x), tab(x) * scalar)
+        assert np.array_equal(result.y, tab.y * scalar)
+
+        # Testing multiplication with bad array
+        x = np.array([1, 2, 3])
+        y = np.array([4, 5, 6])
+        tab = Tabulation(x, y)
+
+        bad_array = np.array([])
+        with self.assertRaises(ValueError) as context:
+            result = tab * bad_array
+        self.assertEqual(
+            str(context.exception),
+            "Cannot multiply Tabulation by given value"
+            )
+
+        # Test in-place multiplication of two Tabulations
+        x1 = np.array([1, 2, 3])
+        y1 = np.array([4, 5, 6])
+        tab1 = Tabulation(x1, y1)
+
+        x2 = np.array([1, 2, 3])
+        y2 = np.array([1, 2, 3])
+        tab2 = Tabulation(x2, y2)
+
+        tab1.__imul__(tab2)
```

Comment:
So `__imul__` is a magic function that is invoked by the `*=` operator. You should really say `tab1 *= tab2`. Fix everywhere. Same for div, add, sub.

Reply:
for some reason, the `__idiv__` magic function isn't working. When I replace it with `\=`, it's defaulting to `__truediv__`. I know this because I'm getting an `AssertionError` from the error messages not matching. It's reading the `__truediv__` error message instead of the `__idiv__` one.

Reply:
Figured it out! `__idiv__` no longer exists. It is now `__itruediv__`. Changing the name of the method fixed it.